services/f-droid/f-droid.tester.js

```diff
@@ -8,165 +8,77 @@ const t = (module.exports = new ServiceTester({
-    * ROOT access only required for extracting paid apps.
-    * Apk's will be saved in /sdcard/Download/Eimon/.
-    * Provided Search option to search applications.
-    * Compatible with latest version of Android 6.0
-    * Saved apk format : AppPackageName.apk.
-
-RepoType: git
-Repo: https://github.com/axxapy/apkExtractor
 
-Builds:
-  - versionName: '1.2'
-    versionCode: 32
-    commit: '0.32'
-    subdir: app
-    gradle:
-      - yes
-
-  - versionName: '1.4'
-    versionCode: 33
-    commit: '5'
-    subdir: app
-    gradle:
-      - yes
-
-AutoUpdateMode: Version %v
-UpdateCheckMode: Tags
-CurrentVersion: 1.4
-CurrentVersionCode: 33
-`
 const base = 'https://gitlab.com'
-const path = '/fdroid/fdroiddata/raw/master/metadata/axp.tool.apkextractor'
-
-t.create('Package is found with default metadata format')
-  .get('/v/axp.tool.apkextractor.json')
-  .intercept(nock => nock(base).get(`${path}.txt`).reply(200, testString))
-  .expectBadge({ label: 'f-droid', message: 'v1.4' })
-
-t.create('Package is found with fallback yml matadata format')
-  .get('/v/axp.tool.apkextractor.json')
-  .intercept(nock => nock(base).get(`${path}.txt`).reply(404))
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(200, testYmlString))
-  .expectBadge({ label: 'f-droid', message: 'v1.4' })
-
-t.create('Trailing 0 in yml format')
-  .get('/v/axp.tool.apkextractor.json')
-  .intercept(nock => nock(base).get(`${path}.txt`).reply(404))
-  .intercept(nock =>
-    nock(base).get(`${path}.yml`).reply(200, "CurrentVersion: '1.4000'")
-  )
-  .expectBadge({ label: 'f-droid', message: 'v1.4000' })
-
-t.create('Package is found with yml matadata format')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(200, testYmlString))
-  .expectBadge({ label: 'f-droid', message: 'v1.4' })
+const path =
+  '/kitsunyan/fdroidapi/raw/master/applications/org.fdroid.fdroid.privileged'
+
+t.create('Package is found')
+  .get('/v/org.fdroid.fdroid.privileged.json')
+  .intercept(nock => nock(base).get(path).reply(200, testJson))
+  .expectBadge({ label: 'f-droid', message: 'v0.2.11' })
+
+t.create('Package is found with latest is suggested')
+  .get('/v/org.fdroid.fdroid.privileged.json?suggested')
+  .intercept(nock => nock(base).get(path).reply(200, testJson))
+  .expectBadge({ label: 'f-droid', message: 'v0.2.7' })
+
+t.create('Package is not found with 403')
+  .get('/v/org.fdroid.fdroid.privileged.json')
+  .intercept(nock => nock(base).get(path).reply(403, 'some 403 text'))
+  .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('Package is not found with "metadata_format" query parameter')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(404))
+t.create('Package is not found with 404')
+  .get('/v/org.fdroid.fdroid.privileged.json')
+  .intercept(nock => nock(base).get(path).reply(404, 'some 403 text'))
```

Comment:
`some 404 text`? Alternatively, you could make this test hit the real API.

Reply:
Yes, missed it when copypasted the test.
I thought it's better to avoid real API because who knows, I chose a package name which doesn't exist, and someone will decide to publish an application with the same package name in the future :eyes:

Reply:
I would argue that there's little risk that someone publishes an app called "io.shiels.does.not.exist", and worse case scenario, we can trivially fix the test. Using the real API at least proves that a 404 is indeed being returned when an app is not found and gives us a little more confidence about the implementation. :)

Reply:
Makes sense. Switched to live test.